backend/app/views.py

```
@@ -625,113 +623,211 @@ def index_view(request, instance=None):
+                require_credentials=True
+            )
 
-                time.sleep(1.5)
+            if form.is_valid():
+                username = form.cleaned_data['username']
+                password = form.cleaned_data['password']
+
+                # Test connection
+                logger.debug(f"Testing auth for {portal.alias} with user {username[0] if username else ''}...")
+                auth_payload = {'username': username, 'password': password, 'url': portal.url}
+                auth_result = utils.try_connection(auth_payload)
```

Ruff (0.13.1)

`utils` may be undefined, or defined from star imports

(F405)